rs/sns/testing/src/sns.rs

```diff
@@ -88,32 +251,199 @@ pub async fn upgrade_sns_controlled_test_canister(
+        .await
+        .unwrap();
+    let _ = propose_and_wait(
+        dev_participant_agent,
+        neuron_id,
+        sns.governance,
+        Proposal {
+            title: "Upgrade SNS controlled canister.".to_string(),
+            summary: "".to_string(),
+            url: "".to_string(),
+            action: Some(Action::UpgradeSnsControlledCanister(
+                UpgradeSnsControlledCanister {
+                    canister_id: Some(canister_id.get()),
+                    new_canister_wasm: modified_test_canister_wasm,
+                    canister_upgrade_arg: Some(Encode!(&upgrade_arg).unwrap()),
+                    mode: Some(CanisterInstallMode::Upgrade as i32),
+                    chunked_canister_wasm: None,
+                },
+            )),
         },
     )
-    .await;
+    .await
+    .unwrap();
+
+    // TODO: @rvem: commented code below relates to the upgrade workflow that uses the
+    // 'upgrade_sns_controlled_canister::exec'
+    // wait_for_proposal_execution(dev_participant_agent, sns.governance, proposal_id)
+    //     .await
+    //     .expect("Failed to execute the proposal");
+
     // Wait for the canister to become available
     await_with_timeout(
-        pocket_ic,
+        dev_participant_agent,
         0..EXPECTED_UPGRADE_DURATION_MAX_SECONDS,
-        |pocket_ic| async {
-            let canister_status = pocket_ic
-                .canister_status(canister_id.into(), Some(sns.root.canister_id.into()))
-                .await;
+        |agent: &C| async {
+            let canister_status = sns
+                .root
+                .get_sns_controlled_canister_status(agent, canister_id)
+                .await
+                .map(|result| result.status);
             canister_status
-                .expect("Canister status is unavailable")
-                .status as u32
+                .map(|status| status == CanisterStatusType::Running)
+                .unwrap_or_default()
         },
-        &(CanisterStatusResultStatus::Running as u32),
+        &true,
     )
     .await
     .expect("Test canister failed to get into the 'Running' state after upgrade");
 }
+
+// Module with PocketIC-specific functions, mainly used in the tests.
+pub mod pocket_ic {
+    use super::{TestCanisterInitArgs, DEFAULT_SWAP_PARTICIPANTS_NUMBER};
+
+    use ::pocket_ic::nonblocking::PocketIc;
+    use candid::Encode;
+    use canister_test::Wasm;
+    use ic_agent::{identity::Secp256k1Identity, Identity};
+    use ic_base_types::{CanisterId, PrincipalId};
+    use ic_nervous_system_agent::{pocketic_impl::PocketIcAgent, sns::Sns};
+    use ic_nervous_system_integration_tests::pocket_ic_helpers::{
+        install_canister_on_subnet, nns::ledger::mint_icp,
+    };
+    use ic_nns_constants::ROOT_CANISTER_ID;
+    use icp_ledger::{Tokens, DEFAULT_TRANSFER_FEE};
+
+    use crate::utils::{swap_participant_secret_keys, NNS_NEURON_ID};
+
+    pub async fn install_test_canister(
```

Comment:
this might be a dumb question, but why can't these functions also work through an Agent interface?

Reply:
Currently, the main use case for `install_test_canister` is to be used in the test-suite.

When it comes to using the `sns-testing` via `bazel run`, users are supposed to install canisters via `dfx`

Reply:
I think it would be good to be consistent with  using the Agent, so that we don't have to rework all these tools when PocketIC eventually becomes something else.

Reply:
Yes, this is a known issue, and I'm discussing it with the PocketIC devs. Anyways, closing this for now, as there's nothing to be done in the scope of this PR.